rules/aip0148/ip_address_format.go

```diff
@@ -0,0 +1,56 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aip0148
+
+import (
+	"strings"
+
+	"github.com/googleapis/api-linter/lint"
+	"github.com/googleapis/api-linter/rules/internal/utils"
+	"github.com/jhump/protoreflect/desc"
+	"google.golang.org/genproto/googleapis/api/annotations"
+)
+
+var ipAddressFormats = []annotations.FieldInfo_Format{
+	annotations.FieldInfo_IPV4,
+	annotations.FieldInfo_IPV6,
+	annotations.FieldInfo_IPV4_OR_IPV6,
+}
+
+var ipAddressFormat = &lint.FieldRule{
+	Name: lint.NewRuleName(148, "ip-address-format"),
+	OnlyIf: func(fd *desc.FieldDescriptor) bool {
```

Comment:
we should also check that the field is of type string, to comply with go/aip/148#well-known-string-fields. I'm not sure if we should throw an error for this for an already existing field, where changing the type is a breaking change. Do we have any precedents for this kind of change to the linter?

Reply:
We already have a linter rule for that guidance in AIP-202 :) https://linter.aip.dev/202/string-only-format

Reply:
Oh I misunderstood the comment. I think that is a good idea, thanks.

Reply:
Done!